### PR TITLE
Implement os.tempDir

### PR DESCRIPTION
**Pull request by Bin-Huang**
I hope I can do some parts. 😊😉

Feedback/ suggestion / ideas appreciated 😺😺

### CODE REVIEW

File: msg.proto [hunk 1]
```diff
@@ -25,6 +25,8 @@ message Msg {
     READ_FILE_SYNC = 11;
     READ_FILE_SYNC_RES = 12;
     WRITE_FILE_SYNC = 13;
+    Temp_Dir = 14;
+    Temp_Dir_RES = 15;
```
Comment (ZanderBrown) [inline msg.proto]: These should be in capitals

**Comment by monkingxue** [comment]
LGTM, but maybe we need to migrate the code from go to cpp/rust.

File: msg.proto [hunk 2]
```diff
@@ -102,10 +102,10 @@ message Msg {
...
   // write_file_sync_perm specified by https://godoc.org/os#FileMode
 
-  // Temp_Dir
+  // TEMP_DIR
   string temp_dir_parent_path = 140;
   string temp_dir_prefix = 141;
 
-  // Temp_Dir_RES
+  // TEMP_DIR_RES
   stirng temp_dir_path = 150;
```
Comment (kitsonk) [inline msg.proto]: `stirng`?

Comment (Bin-Huang) [reply msg.proto]: carelessness... 😅

**Comment by Bin-Huang** [comment]
@monkingxue Just for curiosity, why?🤔😃

**Comment by kevinkassimo** [comment]
@Bin-Huang Check out the `deno2/` README. Ryan briefly mentioned about the reason (mainly, double GC)

**Comment by ry** [comment]
Thanks @Bin-Huang but I will close this.
We will revisit this soon once the runtime is up and running in rust.

**Comment by Bin-Huang** [comment]
I also can write it in rust, waiting for yous😊

**Pull request closed without merging**